bin/spiced/src/lib.rs

```diff
@@ -140,3 +176,46 @@ pub async fn run(args: Args, metrics_handle: Option<PrometheusHandle>) -> Result
         }),
     }
 }
+
+fn load_tls_config(
+    args: &Args,
+) -> std::result::Result<Option<Arc<ServerConfig>>, Box<dyn std::error::Error>> {
+    if !args.tls {
+        return Ok(None);
+    }
+
+    let cert_reader: &mut dyn io::BufRead =
+        match (&args.tls_certificate_file, &args.tls_certificate) {
+            (Some(cert_path), _) => &mut load_file(Path::new(cert_path))?,
+            (_, Some(cert)) => &mut cert.as_bytes(),
+            (None, None) => return Err("TLS certificate is required (--tls-certificate)".into()),
+        };
+    let key_reader: &mut dyn io::BufRead = match (&args.tls_key_file, &args.tls_key) {
+        (Some(key_path), _) => &mut load_file(Path::new(key_path))?,
+        (_, Some(key)) => &mut key.as_bytes(),
```

Comment:
**question:** What loading precedence occurs here, does it follow the position of the match arm? e.g. file ahead of CLI arg, if a user supplies both arguments?

Reply:
Yeah it tries each match arm in order